JS/Lesson19/esra/recursion.js

```diff
@@ -109,6 +141,36 @@ const nestedObj = {
 	},
 };
 
+function singleLevelObj(obj) {
+	const result = {};
+
+	for (const key in obj) {
+		if (typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+		  const flattenedSubObject = singleLevelObj(obj[key]);
+		  for (const subKey in flattenedSubObject) {
+			result[`${key}.${subKey}`] = flattenedSubObject[subKey];
+		  }
+		} else if (Array.isArray(obj[key])) {
+		  result[key] = obj[key].map(item => {
+			if (typeof item === 'object') {
+			  return singleLevelObj(item);
+			} else {
+			  return item;
+			}
+		  });
+		} else {
+		  result[key] = obj[key];
+		}
+	  }
+	
+	  return result;
+	}
+
+
+console.log('Exercise6: ', result);
```

Comment:
Don't forget to run your code and verify that everything works before submitting an assignment :)

```suggestion
console.log('Exercise6: ', singleLevelObj(nestedObj));
```